---

# **MODEL INFERENCE**

In this section, we load the best saved model and use it to generate predictions on new data. The data for inference must be in its original/raw format (i.e., not preprocessed), since the model’s pipeline already includes all preprocessing steps such as scaling and encoding.

In [4]:
# 1. Import library
import pandas as pd
import pickle

# 2. Load pipeline model
with open('best_model_dtc.pkl', 'rb') as file:
    loaded_model = pickle.load(file)

In [5]:
# Example of new input data for inference (original/raw format, without manual preprocessing), for instance, a new website visitor session.
new_df = pd.DataFrame([{
    'administrative': 3,
    'administrative__duration': 120.0,
    'informational': 0,
    'informational__duration': 0.0,
    'product_related': 20,
    'product_related__duration': 500.0,
    'bounce_rates': 0.01,
    'exit_rates': 0.05,
    'page_values': 0.0,
    'special_day': 0.0,
    'month': 'Nov',
    'operating_systems': 2,
    'browser': 2,
    'region': 1,
    'traffic_type': 3,
    'visitor_type': 'Returning_Visitor',
    'weekend': True
}])

In [6]:
# Use the model pipeline to generate predictions.
prediction = loaded_model.predict(new_df)
prediction_proba = loaded_model.predict_proba(new_df)

# Tampilkan hasil prediksi
print("Prediction (0=No Purchase, 1=Purchase):", prediction[0])
print("Purchase Probability:", round(prediction_proba[0][1], 4))

Prediction (0=No Purchase, 1=Purchase): 0
Purchase Probability: 0.3333


> **Insight:**
The model successfully predicts whether the visitor is likely to make a purchase, providing a probability estimate (0.33) that can guide business actions such as targeted promotions or follow-ups..

---